python/cog/server/redis_queue.py

```diff
@@ -347,7 +317,12 @@ def handle_message(
             response["status"] = Status.FAILED
             response["error"] = str(e)
         finally:
-            send_response(response)
+            yield response
+
+            try:
+                input_obj.cleanup()
```

Comment:
This shouldn't still be here, should it?

Reply:
The call to `input_obj.cleanup()` further up is only run if the input validation fails – this one handles the case where validation passed and we actually ran the prediction.